src/charts/grouped-bar.js

```diff
@@ -0,0 +1,988 @@
+     * @type function
+     */
+
+    /**
+     * @typedef GroupedBarChartData
+     * @type {Object}
+     * @property {Object[]} data       All data entries
+     * @property {String} name         Name of the entry
+     * @property {String} group        group of the entry
+     * @property {Number} value        Value of the entry
+     *
+     * @example
+     * {
+     *     'data': [
+     *         {
+     *             "name": "2011-01",
+     *             "group": "Direct",
+     *             "value": 0
+     *         }
+     *     ]
+     * }
+     */
+
+    /**
+     * Grouped Bar Chart reusable API module that allows us
+     * rendering a multi grouped bar and configurable chart.
+     *
+     * @module Grouped-bar
+     * @tutorial grouped-bar
+     * @requires d3-array, d3-axis, d3-color, d3-collection, d3-dispatch, d3-ease,
+     *  d3-interpolate, d3-scale, d3-selection, lodash assign
+     *
+     * @example
+     * let groupedBar = GroupedBar();
+     *
+     * groupedBar
+     *     .width(containerWidth);
+     *
+     * d3Selection.select('.css-selector')
+     *     .datum(dataset.data)
+     *     .call(groupedBar);
+     *
+     */
+    return function module() {
+
+        let margin = {
+                top: 40,
+                right: 30,
+                bottom: 60,
+                left: 70
+            },
+            width = 960,
+            height = 500,
+
+            xScale,
+            xScale2,
+            xAxis,
+            yScale,
+            yScale2,
+            yAxis,
+
+            aspectRatio = null,
+
+            verticalTicks = 5,
+            yTickTextYOffset = -8,
+            yTickTextXOffset = -20,
+
+            numOfVerticalTicks = 5,
+            numOfHorizontalTicks = 5,
+            percentageAxisToMaxRatio = 1,
+
+            colorSchema = colorHelper.colorSchemas.britechartsColorSchema,
+
+            colorScale,
+            categoryColorMap,
+
+            layers,
+
+            ease = d3Ease.easeQuadInOut,
+            horizontal = false,
+
+            svg,
+            chartWidth, chartHeight,
+            data,
+            groups,
+
+            transformedData,
+
+            tooltipThreshold = 480,
+
+            xAxisPadding = {
+                top: 0,
+                left: 0,
+                bottom: 0,
+                right: 0
+            },
+            maxBarNumber = 8,
+
+            animationDelayStep = 20,
+            animationDelays = d3Array.range(animationDelayStep, maxBarNumber * animationDelayStep, animationDelayStep),
+            animationDuration = 1000,
+
+            grid = null,
+
+            nameLabel = 'name',
+            valueLabel = 'value',
+            groupLabel = 'group',
+            nameLabelFormat,
+            valueLabelFormat = NUMBER_FORMAT,
+
+            // getters
+            getName = (data) => data[nameLabel],
+            getValue = (data) => data[valueLabel],
+            getGroup = (data) => data[groupLabel],
```

Comment:
These three should be either:
  getName = ({name}) => name,
  getValue = ({value}) => value,
  ...

Or we should remove them